third_party/java/arcs/build_defs/internal/kotlin_serviceloader_registry.bzl

```diff
@@ -0,0 +1,26 @@
+"""Rules for Kotlin jvm particles."""
+
+def _kotlin_serviceloader_registry(ctx):
+    ctx.actions.expand_template(
+        template = ctx.file._template,
+        output = ctx.outputs.out,
+        substitutions = {
+            "{particles}": ctx.attr.particles,
+        },
+    )
+
+kotlin_serviceloader_registry = rule(
+    implementation = _kotlin_serviceloader_registry,
+    attrs = {
+        "particles": attr.string(doc = "Newline separated list of fully qualified particle classes"),
```

Comment:
This would ideally be `attr.string_list` instead

Reply:
Done